apps/meteor/client/views/omnichannel/directory/chats/hooks/useTagsLabels.tsx

```diff
@@ -1,21 +1,24 @@
 import { useEndpoint } from '@rocket.chat/ui-contexts';
 import { useQuery } from '@tanstack/react-query';
-import { useMemo } from 'react';
+import { useCallback, useMemo } from 'react';
 
 // TODO: Remove this hook when the endpoint is changed
 // to return labels instead of tag id's
-export const useTagsLabels = (initialTags: string[] = []) => {
+export const useTagsLabels = (enabled = true) => {
 	const getTags = useEndpoint('GET', '/v1/livechat/tags');
 	const { data: tagsData, isInitialLoading } = useQuery(['/v1/livechat/tags'], () => getTags({ text: '' }), {
-		enabled: Boolean(initialTags.length),
+		enabled,
 	});
 
 	const labels = useMemo(() => {
 		const { tags = [] } = tagsData || {};
 		return tags.reduce<Record<string, string>>((acc, tag) => ({ ...acc, [tag._id]: tag.name }), {});
 	}, [tagsData]);
 
-	return useMemo(() => {
-		return isInitialLoading ? initialTags : initialTags.map((tag) => labels[tag] || tag);
-	}, [initialTags, isInitialLoading, labels]);
+	return useCallback(
+		(tagId: string) => {
+			return isInitialLoading ? tagId : labels[tagId] || tagId;
```

Comment:
Is it returning a tag ID a valid callback here? Isn't better to render another informative text, render nothing, or even throwing an exception?

Reply:
We have to support custom tags which are not returned by `/tags` endpoint. We plan on revisit that tags feature soon, so for now I believe this is acceptable.